src/actions/index.js

```diff
@@ -0,0 +1,68 @@
+import { add, remove, update, getAll } from '../common/helpers/indexedDB';
+import { actionTypes } from '../common/constants/actionTypes';
+const uuidv1 = require('uuid/v1');
+
+export const sendRequestToAddPost = post => async dispatch => {
+  dispatch(isRequestPending(true));
+  post = {
+    ...post,
+    id: uuidv1()
+  };
+  await add(post);
```

Comment:
Please, wrap all async requests in `try ... catch` block and dispatch ..._FAILURE action if something is wrong. Read more about error handling